apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#region
+
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+#endregion
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    public class BindingTests : IClassFixture<TestDatabaseFixture>
+    {
+        public BindingTests(TestDatabaseFixture testFixture)
+        {
+            _testFixture = testFixture;
+        }
+
+        private readonly TestDatabaseFixture _testFixture;
+
+        private async Task TestBind<T>(SpannerDbType parameterType, T value)
+        {
+            int rowsRead;
+            T valueRead = default(T);
+
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var cmd = connection.CreateSelectCommand("SELECT @v",
+                    new SpannerParameterCollection {new SpannerParameter("v", parameterType)});
+
+                cmd.Parameters["v"].Value = value;
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    rowsRead = 0;
+                    while (await reader.ReadAsync())
+                    {
+                        valueRead = reader.GetFieldValue<T>(0);
+                        rowsRead++;
+                    }
+                }
+            }
+            Assert.Equal(1, rowsRead);
+            Array valueAsArray = value as Array;
+            if (valueAsArray != null)
+            {
+                Array valueReadAsArray = valueRead as Array;
```

Comment:
I don't think it's Array that's the problem - it's that it's a generic type parameter. Do you actually need the method to be generic at all? If you make it non-generic and make the type of `value` just `object`, I suspect all of this will go away. The only tricky part is `GetFieldValue<T>`... you could potentially make it still a generic method, but have `value` and `valueRead` as `object`...
(Anyway, it's not a problem as you've got it at the moment. Just thinking of options.)